controllers/volumereplicationgroup_controller.go

```diff
@@ -1476,36 +1500,48 @@ func (v *VRGInstance) PVUploadToObjectStore(pvc *corev1.PersistentVolumeClaim,
 	return s3Profiles, nil
 }
 
+func (v *VRGInstance) getOrCacheObjectStorer(s3ProfileName string) (ObjectStorer, error) {
+	if v.objectStorers == nil {
+		v.objectStorers = make(map[string]CachedObjectStorer)
+	}
+
+	// return a cached objectStore if present
+	if cachedObjectStore, ok := v.objectStorers[s3ProfileName]; ok {
+		return cachedObjectStore.storer, cachedObjectStore.err
+	}
+
+	// create a new objectStore (connection)
+	objectStore, err := v.reconciler.ObjStoreGetter.ObjectStore(
+		v.ctx,
+		v.reconciler.APIReader,
+		s3ProfileName,
+		v.namespacedName,
+		v.log)
+	if err != nil {
+		err = fmt.Errorf("error connecting to object store for s3Profile %s, %w", s3ProfileName, err)
+	}
+
+	// cache the objectStore (or the error)
+	v.objectStorers[s3ProfileName] = CachedObjectStorer{
+		storer: objectStore,
```

Comment:
If this objectstore encounters an error then it is doomed forever, right?  Once it is added with an error, we will always return it in line 1510 with the same error.  No retry.

Reply:
Which also makes me wonder why we do not close the sessions we open with the object store? (different issue, but was checking based on this comment if we were closing the connection which would cause other failures)